scripts/ruby_common/daily_snapshot.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require_relative 'slack_client'
+require_relative 'docker_utils'
+require 'rubygems'
+require 'logger'
+require 'fileutils'
+require 'active_support/time'
+
+# Retrieves an environmental variable, failing if its not set or empty.
+def get_and_assert_env_variable(name)
+  var = ENV[name]
+  raise "Please set #{name} environmental variable" if var.nil? || var.empty?
+
+  var
+end
+
+BASE_FOLDER = get_and_assert_env_variable 'BASE_FOLDER'
+SLACK_TOKEN = get_and_assert_env_variable 'SLACK_API_TOKEN'
+CHANNEL = get_and_assert_env_variable 'SLACK_NOTIF_CHANNEL'
+LOG_DIR = './'
+
+CHAIN_NAME = ARGV[0]
+raise 'No chain name supplied. Please provide chain identifier, e.g. calibnet or mainnet' if ARGV.empty?
+
+# Current datetime, to append to the log files
+DATE = Time.new.strftime '%FT%H:%M:%S'
+LOG_EXPORT = "#{LOG_DIR}/#{CHAIN_NAME}_#{DATE}_export"
+
+# Create log directory
+FileUtils.mkdir_p LOG_DIR
+
+loop do
+  client = SlackClient.new CHANNEL, SLACK_TOKEN
+
+  # Find the snapshot with the most recent modification date
+  LATEST = Dir.glob("#{BASE_FOLDER}/s3/#{CHAIN_NAME}/*").max_by {|f| File.mtime(f)}
+
+
+  # Check if the date of the most recent snapshot is today
+  if Time.new.strftime('%F') == File.stat(LATEST).mtime.strftime('%F')
```

Comment:
converting to `Date` and comparing would be a bit more readable than `%F`s. You can use `.to_date` method of `Time`.